src/c++/simulator/contact.cpp

```diff
@@ -0,0 +1,51 @@
+﻿#include <simulator/contact.hpp>
+
+ContactPair::ContactPair(SphericParticle& i, SphericParticle& j) 
+{
+	particles.push_back(std::make_shared<SphericParticle>(i));
```

Comment:
This method and the one below should probably be deleted, as they may be dangerous

Reply:
How about using `reserve` function in the constructor and keep the `push_back`?

Reply:
I was not referring to the vector, but to the copy. I don't find it reasonable to pass a particle by reference only to copy it and then ignoring the original one. I'd expect that the particle that I'm sending to the ContactPair to be the same as the one I'm maintaining in the particles vector.

If in doubt, check this: https://godbolt.org/z/fdr3WE7js

Reply:
Oh, yes. My bad. I didn't notice the `std::make_shared`. I totally agree with you.